app/apollo/resolvers/common.js

```diff
@@ -122,6 +145,24 @@ const getAllowedGroups = async (me, org_id, action, field, queryName, context) =
   return allowedGroups;
 };
 
+// return user permitted cluster groups in an array
+const filterGroupsToAllowed = async (me, org_id, action, field, groups, context) => {
+  const {models} = context;
+  var objectArray = groups.map(group => {
+    return {
+      type: TYPES.GROUP,
+      action,
+      uuid: group.uuid,
+      name: group.name
+    };
+  });
+  var decisions = await models.User.isAuthorizedBatch(me, org_id, objectArray, context);
+  groups = _.filter(groups, (val, idx)=>{
+    return decisions[idx];
+  });
+  return groups;
+};
+
```

Comment:
getAllowedGroups and filterGroupsToAllowed largely duplicate each other.  Consolidate onto just one.

Reply:
removed filtergroupstoallowed as it was unused.